### DISCUSSION TITLE
override colours for blockquote notices

### DISCUSSION THREAD

**Discussion by Andre601** [mmistakes/minimal-mistakes#2995] [Q&A] [answered]
When using the `{: .notice--warning }` option in a Blockquote did I spot something rather bad.
While the colour of the background was set to the warning colour (Which I would like to adjust too) was the line on the left still the default colour of the blockquote, which makes it look very out of place.

How would I go about updating this and possibly some other colours to make block quotes have the same colour as the notice they got added to them?

**Comment by mmistakes** [comment, marked as answer]
https://github.com/mmistakes/minimal-mistakes/blob/master/_sass/minimal-mistakes/_notices.scss

**Reply by Andre601** [reply]
Doesn't answer much to be honest here... Like I don't really know what parts I would need to edit, nor how I would need to edit it.
You can expect me here to be an extreme (S)CSS noob.

**Reply by mmistakes** [reply]
To customize the theme you need to know CSS and or Sass. There are plenty of tutorials out there to get you started.

**Reply by Andre601** [reply]
Yeah sure... Tutorials exist and all, but in all honesty is this one of if not the most lazy response to give.
"Oh. You want to know how to do this?" \*proceeds to just paste URL without context*

I understand that it takes valuable time from you to answer things, but just throwing out a URL to a file and expecting people to instantly know how they can apply changes without messing up other parts of the theme is a lot to ask for here.
I'll try to research the ins and outs of Sass now, but honestly is my joy of using this jekyll theme hurt by this lack of help here... Thanks anyways.

**Reply by mmistakes** [reply]
Sorry. Assumed you read the documentation covering how to modify the theme styles first.

https://mmistakes.github.io/minimal-mistakes/docs/stylesheets/#customizing

And/or searched through the discussions/issues where I’ve answered similar questions before about changing colors and styles.

**Reply by Andre601** [reply]
Yeah... I had a hard time finding it to be honest... And the search on discussions and issue isn't really the best if you ask me...

**Reply by Andre601** [reply]
So if I read this page correctly would I need to copy the entirety of the `_sass` folder into my own setup just to update this one border colour? (Jekyll really sucks for this tbh...)

I personally see this as a sort of bug, because I kinda doubt that it is intentional, that a `notice--warning` blockquote should show a bright blue `border-left` line.

**Reply by iBug** [reply]
@Andre601 Just in case you aren't aware already, it's a *courtesy* for an open-source project maintainer to answer support questions. Coding and maintaining is fun, but providing support is not. In fact, no one is obligated to provide any free support.

If you consider yourself "a complete noob", then a better thing to do is you get yourself educated so you're not "a complete noob" anymore. By asking a support question here, you're expecting others to spend *their* time to help you. Then why shouldn't *you* help others save *their* time by skipping the all-around-the-Internet tutorials and learning them on yourself? As the old saying goes, *One good turn deserves another*.

I'll quote [Daniel Stenberg's answer](//stackoverflow.com/a/55885729/5958455). Danial is *the* maintainer of the popular cURL library for two decades. Emphasis mine.

> 4. curl is free **but my time is not**. I still have a job and someone still has to pay someone for me to get paid every month so that I can put food on the table for my family. I charge customers and companies to help them with curl.

**Reply by Andre601** [reply]
> @Andre601 Just in case you aren't aware already, it's a _courtesy_ for an open-source project maintainer to answer support questions. Coding and maintaining is fun, but providing support is not. In fact, no one is obligated to provide any free support.
> 
> If you consider yourself "a complete noob", then a better thing to do is you get yourself educated so you're not "a complete noob" anymore. By asking a support question here, you're expecting others to spend _their_ time to help you. Then why shouldn't _you_ help others save _their_ time by skipping the all-around-the-Internet tutorials and learning them on yourself? As the old saying goes, _One good turn deserves another_.

I am more than aware of it. After all do I spend my time on various open source projects too.
While I can agree that time for a maintainer is limited, do I not agree that a simple posting of a SCSS file link is enough to give someone any clue. Only if that person had his fair share of work regarding SCSS (which believe it or not is still not that common nor popular to use compared to common CSS) would it be understandable that they may have the knowledge to work on this.
But even then is it not guaranteed, since there is no guarantee that the user has the required knowledge about Ruby, Jekyll and Mmistakes to handle all issues on their own.

My personal approach on this is to treat anyone as a beginner, unless you know they have their knowledge.

I mean what gives a better impression as an answer?

1. Hey there. Please take a look at https://github.com/mmistakes/minimal-mistakes/blob/master/_sass/minimal-mistakes/_notices.scss for all the variables used for notices and https://mmistakes.github.io/minimal-mistakes/docs/stylesheets/#customizing for how to customize things in mistakes.

or
2. https://github.com/mmistakes/minimal-mistakes/blob/master/_sass/minimal-mistakes/_notices.scss

I personally would go with option one as it right from the start gives enough info about where to look for what.

Again: I know support is time-consuming and that it's not a privilege for users of an open-source project, but you should still go for a certain level of quality for support here.
And if time really isn't an available source, then **find people**. Encourage active members of the community to share their solutions and let them be more integrated into a project. It can save you a lot of time and maybe even prevent you from a support burn-out.

I also made a Blog post once with a section pointing out the common pitfall of people assuming that someone seeking support already knows everything there is to know:
https://www.andre601.ch/blog/posts/2021/06-26-critique-vs-hate/#wrong-expectations

**Reply by iBug** [reply]
Glad that you understand OSS maintainer's difficulties in incoming support questions. That's an invaluable quality.

Though it seems that you still have some misunderstanding of my idea. For example, you said

> but you should still go for a certain level of quality for support here.

Let me reiterate: It's a *courtesy* to provide support, let alone *quality* support. Providing support, among all tasks in an open-source project, is the most time-consuming, yet the least rewarding. And most maintainers don't live off their OSS, unlike Martin Donath (Material for MkDocs), whose insider model also generated him enough income to make a living. Michael OTOH, as far as I can tell, has his own living and limited time to spend on this theme.
There are also less active contributors to this project, compared to Material for MkDocs, despite some more stargazers (8k vs 7k). Given the primary audience of the projects (those getting a nice-looking personal website vs. those trying to produce documentation), it could be harder to find people with the appropriate / adequate knowledge to handle support questions. And that turns the question back again: Why would people spend their time providing support for no obvious reward?

Tt's easier said than done, and that's why you could argue, from your perspective, that every OSS should provide support to users. That's because you're asking, not giving. Most noob-level questions have been asked and answered for times and times, and it only becomes more and more boring when an extra one comes up. People have patience, and it drains. I once asked Michael why he wouldn't make some FAQ (#2397), and his response was:

> Not sure how to handle those as I don't really want to be a "how to use Jekyll guide"...

Pretty much the same. I sympathize with Michael that most questions are really about using Jekyll. Yours (this one) is no different. Michael gave you the direction (which specific CSS to change / override), and everything beyond that is not theme-specific but general to Jekyll, which is also the point where it starts to become boring, if not burdensome.

The bottom line is: Do not demand anything from sources provided free of charge.

**Reply by Andre601** [reply]
Those are some good points there.
But I also notice a common flaw in those thoughts, most noticeably this question:

> Why would people spend their time providing support for no obvious reward?

If you ask yourself this question, then it's honestly best to step away from the project(s) as a whole. Support never was and never will be about getting a reward. The best reward you can get from it is the knowledge of a satisfied and happy user/customer, but you can't expect to get a donation or even "thanks" for every support request you solved.

It's a curse of OS projects: Lot of work for little to no reward. Like I could rephrase the above question like this:

> Why would people spend their time providing bug fixes and feature implementations for no obvious reward?

This is the essential issue with various Maintainers of OS projects. They often have a mindset of "I won't implement it if there is no personal gain from it".
Sure, they spend time on this stuff, but if you really go on implementing features or even bug fixes only once you see some reward/personal gain for you, then you should honestly quit OS and instead look for a full-time job instead.

Also, on another note, not having an FAQ is imo a stupid approach. the goal of an FAQ is to obviously answer common questions. This would also be one of the only few cases where I would be fine with getting just a URL provided if it points to the FAQ entry in question.
An FAQ aims to ease support in different ways by, as the name suggests, answering frequently asked questions, so the approach of not having an FAQ because you "don't want to be a how-to-use Jekyll guide" is kind of weird.

In general, the goal of an OS Project is at first not to make any profit nor gain any real rewards outside of random donations and code contributions. The main focus should be to bring something to the community. If you approach Support requests only with the thought if it's worth it without any reward, then you misunderstand the OS concept as a whole imo.

Anyway, this post has become somewhat off-topic now... Sorry about that.

**Reply by iBug** [reply]
I see where you come from. Most OSS all boil down to one thing: Whether one likes doing it, or not. Providing support could, or could not, be enjoyable to one. It's like you're implying that "answering support questions is an obligation for maintainer", which is often not, and not polite as well. Answering support questions is not equivalent to making a project easier to use and more reachable in general, specifically given their difference in forms of carrying out and the resulting audience.

BTW, Now I have an extra supporter for the "make an FAQ" proposal `:)`

Anyway, let's call this a conclusion and stop going further off-topic. I value your conscious and detailed thoughts. It's been a nice experience exchanging ideas with you. Have a nice day.

**Reply by mmistakes** [reply]
Thanks for jumping in @iBug. You've seen enough of my responses to topics like this to know what I would end up saying.

@Andre601 Given the options, would you have preferred?
1. Me leaving a link to the source code that controls the notice styling.
2. Not replying at all.

My mind with all this is:
- I don't approach answering support requests as a way of making a profit or reward. It is something I do to help the community. Just look at the [Jekyll Talk forums](http://talk.jekyllrb.com/) where I'm very active as well. I think you'd be hard pressed to find other actively maintained Jekyll themes with the level of free support I provide. Even the core Jekyll ones are ghost towns...
- I started this project (and my other Jekyll theme repos) as a way of sharing my site's design and work with the community.
- The "theme" was and is provided as is. If you want to deviate from what's included it's expected you'll research how to do that.
- Jekyll requires a level of web development knowledge. After all it's a command line tool. I wouldn't expect a "noob" to use it unless they want to level-up their skills.
- It is not my responsibility to teach Jekyll or how to build web sites. When I have the time I do try to do that, but when a topic has been covered numerous times before it puts the burden on me. When most of the time searching through closed issues would surface an answer.
- This is a popular project with a ton of users... and I make next to nothing maintaining and answering questions about it. @iBug is pretty much the only user who reliably tries to help field support questions. Others have jumped in over the years, but it's usually self motivated as a way of pushing through a PR they want me to green light.

Perhaps this is a failing on my part and I should have never turned on Issues and Discussions and stated in the `README` that you're on your own when using this theme.

OSS is hard.
